src/core/server/deprecations/deprecations_route_handler_context.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreElasticsearchRouteHandlerContext } from '../elasticsearch';
+import type { CoreSavedObjectsRouteHandlerContext } from '../saved_objects';
+import type { DeprecationsClient, InternalDeprecationsServiceStart } from './deprecations_service';
+
+/**
+ * Core's `deprecations` request handler context.
+ * @public
+ */
+export interface DeprecationsRequestHandlerContext {
+  client: DeprecationsClient;
+}
+
+/**
+ * The {@link DeprecationsRequestHandlerContext} implementation.
+ * @internal
+ */
+export class CoreDeprecationsRouteHandlerContext implements DeprecationsRequestHandlerContext {
+  #client?: DeprecationsClient;
```

Comment:
For some reason, this is the only part I don't particularly like since I'm scratching my head to think of particular situations where the client wouldn't be available. It's just a feeling though, nothing holding the PR back.